pointcloud/basic_octree_utils.go

```diff
@@ -99,9 +99,9 @@ func (octree *BasicOctree) splitIntoOctants() error {
 
 // Checks that a point should be inside a basic octree based on its center and defined side length.
 func (octree *BasicOctree) checkPointPlacement(p r3.Vector) bool {
-	return ((math.Abs(octree.center.X-p.X) <= octree.sideLength/2.) &&
-		(math.Abs(octree.center.Y-p.Y) <= octree.sideLength/2.) &&
-		(math.Abs(octree.center.Z-p.Z) <= octree.sideLength/2.))
+	return ((math.Abs(octree.center.X-p.X) <= (1+nodeRegionOverlap)*octree.sideLength/2.) &&
+		(math.Abs(octree.center.Y-p.Y) <= (1+nodeRegionOverlap)*octree.sideLength/2.) &&
+		(math.Abs(octree.center.Z-p.Z) <= (1+nodeRegionOverlap)*octree.sideLength/2.))
```

Comment:
This approach lgtm and matches what is generally recommended for working with floats!

From https://learn.microsoft.com/en-us/cpp/build/why-floating-point-numbers-may-lose-precision?view=msvc-170:
> To resolve the behavior, most programmers either ensure that the value is greater or less than what is needed